packages/@react-stately/combobox/src/useComboBoxState.ts

```diff
@@ -306,33 +307,39 @@ export function useComboBoxState<T extends object>(props: ComboBoxStateOptions<T
     setFocusedState(isFocused);
   };
 
-  let selectionManagerProxy = useMemo(() => {
-    let proxy = new Proxy(selectionManager, {
-      get(target, prop, receiver) {
-        // If the menu is closed, don't update the selected key.
-        if (prop === 'replaceSelection' && !triggerState.isOpen) {
-          return () => {};
-        }
-        return Reflect.get(target, prop, receiver);
-      }
-    });
+  useEffect(() => {
+    // Track the last filtered collection while the menu is open so that we can freeze the displayed collection when the
+    // user clicks outside the popover to close the menu. Prevents the menu contents from updating as the menu closes.
+    if (triggerState.isOpen) {
+      setLastCollection(filteredCollection);
```

Comment:
unfortunately, now we're creating an extra render cycle
is there any way to save the state in an event instead of in an effect?

Reply:
Not that I can think of because of the async case I mentioned [here](https://github.com/adobe/react-spectrum/pull/4060#discussion_r1106143505) where `items` would be changing separate from a event triggered by the user:
1. User types in Luke in combobox -> triggers rerender due to input value changing -> we can update last collection via event
2. useAsyncList finishes fetching the new list -> triggers rerender due to new `items` prop -> need to update lastCollection as a sideeffect of filteredCollection updating as a result of this prop change, no event trigger to watch for from within the combobox